src/ClientConfig.cc

```diff
@@ -325,12 +326,21 @@ bool ClientConfig::LoadConfig(const std::string &_file)
           {
             // Sanity check: Make sure that the server is not already stored.
             bool repeated = false;
-            for (auto const savedServer : this->Servers())
+            for (auto & savedServer : this->Servers())
             {
               if (savedServer.Url().Str() == serverURL)
               {
-                ignwarn << "URL [" << serverURL << "] already exists. "
-                       << "Ignoring server" << std::endl;
```

Comment:
I think this warning should be printed even if you set the key, because the `repeated` flag will be true